controller/lib/hal/hal.h

```diff
@@ -28,12 +28,34 @@
 // when compiling for test, but not when compiling for production.
 
 #ifdef TEST_MODE
+
+#ifdef AVR
+#error "TEST_MODE intended to be run only on native, but AVR is defined"
+#endif
```

Comment:
FYI the reason I had to do TEST_MODE is because platformio's library detection code isn't aware of -DAVR, and so it was trying to compile googletest on AVR and that did not go well.